### ISSUE TITLE
oog Vorbis Tags

### ISSUE THREAD

**Issue by ktnlikeasmallcat** [Beinsezii/ompl#3]
Hi there me again. There's hopefully some helpful information here: 

Ompl 0.5.1 and the newest push using symphonia don't seem to be reading certain newly changed .oog tags, regardless of purging and re-adding them. These tracks do not play in ompl but instead skip through quickly in the now playing UI, I might guess because the file they refer to is not a file anymore, but a file that has been changed, though there are not "duplicates". These tracks play in Clementine.

The output of `ompl print track` while it's iterating through non-playing tracks is as follows, it just plucks the track it's on in that frame and displays its old info.
```
/home/ktn/Music/Aphex_Twin/RDJ/Aphex Twin - RDJ - girl-boy (18 snare rush mix).ogg
GAIN: 1

artist: jnky
date: 20100710
description: https://www.youtube.com/watch?v=WHB9cNflDyg
encoder: Lavc58.134.100 libvorbis
language: eng
purl: https://www.youtube.com/watch?v=WHB9cNflDyg
title: girl-boy (18 snare rush mix)
tracknumber: 13
```

Strangely enough, the inability to change tags does not affect all oog files. I can update the tag of some other files and it will appear changed in ompl, but after it's altered, it will not play.

For example in the applications I'm variously using to change these tags - Clementine, Musicbrainz Picard, or EasyTAG, the track name for the file `~/Music/Aphex_Twin/Windowlicker.oog` is "Windowlicker" however, ompl displays the tag from the originally downloaded and parsed "Windowlicker「HQ」".

Similarly, the artist name for 'The RDJ Album' in other tag readers & players are correct, but in ompl the artist name for 'Aphex Twin - RDJ Album' is "jnky". (I was hoping there was something going on with the $ character in the track "inkey$" but changing this had no effect, and also did not update the track title in ompl. IDK where it's getting "jnky" but I can't seem to change it.)

Other examples of misread tags are found in a "???" section in the genre pane, all basically conforming to the pattern of not reading  newly updated oog tags (I can give other examples if it's useful) and not playing the track, either through using `ompl main` or wrapper scripts, and either purged and re-appended through both the TUI and CLI commands. 

Thanks a lot for your time.

**Comment by Beinsezii** [comment]
Interesting. You mentioned purging and re-adding, but does this also still happen when you restart the UI? Funnily enough I didn't really change anything about the .ogg tag reader; it should behave the same.

The tags are all read in one big batch when you add a library. They don't update or anything when playing. If there's an issue during playback it's cause the file couldn't be read for whatever number of reasons.



As for the different tag values showing up in Clementine vs mine, there's two possibilities:

1) if it's an MP3 file, it has non-id3 tags present that get read by Clementine. I only support ID3 on mp3 files as it's the defacto standard nowadays. Most tag editors will convert APE or whatever to ID3.

2) When reading tag history I use `.skip_to_latest()` for what I assume is the most recent commit. It's possible I need to use ".current()" instead, but Symphonia's documentation doesn't really explain it well. If you can attach a problem file in a tarball I could debug this better.

**Comment by ktnlikeasmallcat** [comment]
It does still happen when restarting. (Also just figured out "jnky" comes from the channel name, a la youtube-dlp.) 

[windowlicker.tar.gz](https://github.com/Beinsezii/ompl/files/10172260/windowlicker.tar.gz)

**Comment by ktnlikeasmallcat** [comment]
Here's a ~~workaround~~: Converting problem files to mp3 doesn't immediately fix the tags. However, when converting the oog to mp3, the id3 tags still show the wrong information. Changing the id3 tags in the new mp3 file fixes the whole issue, and converting those mp3 files back to oog results in the correct tags, as well as the ability to edit those tags and reimport the library without problems. This seems to be actually a problem with my youtube-dlp config converting things to oog, and not a problem in ompl.
...
The problem was as far as I can tell the line `--embed-metadata` in my youtube-dlp config, as changing that and redownloading the files does not present any of the original issues, so I'll close this issue.

**Comment by Beinsezii** [comment]
Just a heads up, this *was* actually an issue with OMPL it seems. Both VLC && Ex Falso showed different tags with the provided tracks, and using the tag history reading change I mentioned earlier fixed it: [6141beb](https://github.com/Beinsezii/ompl/commit/6141bebe4b1cecd3c22291ae33370c2f6c7a4286)

Biiit late but yeah no need to re-encode your stuffs.
